src/app/ipv/middleware.js

```diff
@@ -245,7 +250,11 @@ module.exports = {
         );
 
         req.session.currentPage = "pyi-attempt-recovery";
-        return res.redirect(req.session.currentPage);
+        return await saveSessionAndRedirect(
+          req,
+          res,
+          `/ipv/page/pyi-attempt-recovery`
```

Comment:
This might conflict with https://github.com/govuk-one-login/ipv-core-front/pull/1102/files - once that's been merged, you might find you can simplify this, as we'll only redirect in a couple of places

cc @shivanshuit914